packages/react-positioning/src/middleware/shift.ts

```diff
@@ -0,0 +1,23 @@
+import { shift as middleware, limitShift } from '@floating-ui/dom';
+import { Boundary } from '../types';
+import { getBoundary } from '../utils/getBoundary';
+
+export interface ShiftMiddlewareOptions {
+  hasScrollableElement?: boolean;
+  disableTether?: boolean | 'all';
+  overflowBoundary?: Boundary;
+  container: HTMLElement | null;
+}
+
+export function shift(options: ShiftMiddlewareOptions) {
```

Comment:
Can you please add a JSDoc why we need this?

Reply:
added